src/wasm-vm/contract.ml

```diff
@@ -0,0 +1,21 @@
+open Wasm
+
+type t = {
+  module_ : Ast.module_;
+  storage : bytes;
+}
+
+let make ~storage ~code =
+  try
+    let module_ =
+      match Parse.string_to_module code with
+      | { it = Script.Textual module_; _ } ->
+        Valid.check_module module_;
+        module_
+      | _ -> assert false in
```

Comment:
We shouldn't crash the node when user submits binary repr of the module

Reply:
If you observer the parser we can't get into the binary and quoted formats from `string_to_module`.

https://github.com/Zett98/spec/blob/deku/text/parser.mly#L1173

I also added a test case if you'd like to check for yourself, or if I'd understanding this incorrectly.

```ocaml

let test_successful_parsing () =
  let code = "(module binary \"\x00asm\x01\x00\x00\x00\x01\x07\x01`\x02\x7F\x7F\x01\x7F\x03\x02\x01\x00\x05\x03\x01\x00\x00\x07\x10\x02\x03add\x00\x00\x06memory\x02\x00\n\t\x01\x07\x00 \x00 \x01j\x0B\x00$\x10sourceMappingURL\x12./release.wasm.map\")"
  in
  ignore (parse code)
```